pkg/pillar/cmd/zedrouter/zedrouter.go

```diff
@@ -215,8 +222,19 @@ func (z *zedrouter) init() (err error) {
 	gcp := *types.DefaultConfigItemValueMap()
 	z.appContainerStatsInterval = gcp.GlobalValueInt(types.AppContainerStatsInterval)
 
+	content, err := os.ReadFile(types.ServerFileName)
 	if err != nil {
-		return err
+		return fmt.Errorf("failed to read the controller hostname from file %s: %v",
+			types.ServerFileName, err)
+	}
+	z.controllerHostname = string(content)
```

Comment:
For some of us, using local cluster, we do port mapping on the server, and the server file has <hostname:port> syntax. May need to detect if this type of syntax is used in the server file.

Reply:
@naiming-zededa looks like that is handled below using SplitHostPort.

Reply:
Ok